### DISCUSSION TITLE
How to get the count of HATCH boundary paths?

### DISCUSSION THREAD

**Discussion by ghost** [mozman/ezdxf#455] [Q&A]
Hi guys,

Is there a way to find Loops count and to know if this is a Outer or Inner Loop?

**Comment by mozman** [comment]
Getting the count of boundary paths is the easy part: `len(hatch.paths)`

The nesting structure is more complicated:

1. A hatch can contain multiple geometrically independent areas in a single `HATCH` entity.
2. There is no reliable DXF attribute which indicates the geometrical meaning of a path like exterior or interior path
3. Paths can be deeply nested 

There is a helper tool to create a complex recursive data structure which represents the nesting structure of `HATCH` boundary paths: `ezdxf.path.fast_bbox_detection()`, see the [source code](https://github.com/mozman/ezdxf/blob/master/src/ezdxf/path/nesting.py).

The `fast_bbox_detection()` function expects the boundary path as `ezdxf.path.Path` objects, the factory function `ezdxf.path.from_hatch()` converts all `HATCH` boundary paths into such `Path` objects, but the association to the source DXF boundary path get lost!

An example for `HATCH` processing is implemented in the drawing add-on, see the [draw_hatch_entity()](https://github.com/mozman/ezdxf/blob/80c9c8b4aaaefbcbabd0e5b1697efae2cc3ce26a/src/ezdxf/addons/drawing/frontend.py#L333) method in source code of the `Frontend()` class. Note: most backends require a specific winding orientation of the paths to represent filled areas or holes.
